weblate/utils/requirements.py

```diff
@@ -169,10 +170,81 @@ def get_versions():
     return result
 
 
+def get_version_number_db(database, version):
+    """Returns version of database in format as (Major Version.Minor Version.Revision).
+
+    from an integer. To know more about database version,
+    refer <https://www.psycopg.org/docs/connection.html>.
+    """
+    if database == "POSTGRESQL":
+        major_version = version[0:2]
+        minor_version = version[2:4] if version[2:4] != "00" else ""
+        revision = version[5:]
+        return (
+            major_version + "." + minor_version + "." + revision
+            if minor_version != ""
+            else major_version + "." + revision
+        )
+    else:
+        return str(version[0]) + "." + str(version[1])
+
+
+def get_db_version():
+    if connection.vendor.upper() == "POSTGRESQL":
+        try:
+            result = (
+                connection.vendor,
+                "https://www.postgresql.org/",
+                get_version_number_db(
+                    connection.vendor,
+                    str(connection.cursor().connection.server_version),
+                ),
+            )
+        except RuntimeError:
+            raise ImproperlyConfigured(
+                "Failed to get a database version. please install a database."
+            )
+    else:
+        try:
+            result = (
+                connection.vendor,
+                "https://www.mysql.com/",
+                connection.cursor().connection.get_server_info().split("-", 1)[0],
+            )
+        except RuntimeError:
+            raise ImproperlyConfigured(
+                "Failed to get a database version. please install a database."
+            )
+
+    return result
+
+
+def get_cache_version():
+    try:
+        import os
+
+        cmd = "redis-server --version"
+        result = (
+            "Redis",
+            "https://redis.io/",
+            str(os.popen(cmd).read().split(" ")[2][2:]),
```

Comment:
There is no reason to have redis installed and running on same machine. Actually, in all Docker deployments it is not installed.

What should work instead (after checking that cache backend is redis) is: `cache.client.get_client().info()['redis_version']`

Reply:
Hi @nijel , 
I am checking cache backend as Redis through `settings.CACHES.get('default').get('BACKEND') == "django_redis.cache.RedisCache"`.
Also, I'm not sure what to import  for `cache.client.get_client().info()['redis_version']` to work??
I have tried checking with `import django_redis.cache` but it's throwing some error or may I'm importing something wrong.

Reply:
`from django.core.cache import cache` should do it.

Reply:
Yes, it works. Thanks

Reply:
@nijel - the code `from django.core.cache import cache` looks magical. Can you refer a source where its available so that I can refer it for the future?

Reply:
https://docs.djangoproject.com/en/3.0/topics/cache/#the-low-level-cache-api

Reply:
@nijel - Thanks for sharing. I read it over the weekend here.